src/main/java/org/springframework/data/elasticsearch/core/query/DeleteQuery.java

```diff
@@ -0,0 +1,468 @@
+/*
+ * Copyright 2013-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.core.query;
+
+import co.elastic.clients.elasticsearch._types.Conflicts;
+import co.elastic.clients.elasticsearch._types.Slices;
+import co.elastic.clients.elasticsearch._types.query_dsl.Operator;
```

Comment:
We cannot use Elasticsearch classes outside of the `'org.springframework.data.elasticsearch.client.elc` package, we have archunit tests that fail, they ensure that.
In the `DeleteQuery` we only can use our own classes and then map them to Elasticsearch classes in the request converter.

_don't know why this comment wasn't taken into my review at first try_

Reply:
I also believe that there are several common fields among all queries, such as the lucene query `q` and its parameters, the `explain` option, the `expandWildcards` option, and more. However, they cannot be independent as it is a matter of design later on.